app/controllers/campaigns_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CampaignsController < ApplicationController
+	before_action :set_campaign, only: [:show, :edit, :update]
+
+	def index
+		@campaigns = Campaign.all
+	end
+
+	def create
+		@campaign = Campaign.new(campaign_params)
+		if @campaign.save
+			redirect_to @campaign
+		else
+			render 'new'
```

Comment:
Rather than rendering from routes like /create and /update, you probably want to redirect_to new_campaign_path / redirect_to edit_campaign_path(@campaign), I think?